### PR TITLE
Rename variables in Base64.* for readability

### PR DESCRIPTION
**Pull request by Joe-Abraham**
Follow-up PR: https://github.com/facebookincubator/velox/pull/10371

The following changes are done in this PR.
- Variable names are updated for better readability.
- Function declarations in Base64.h are reordered for better maintainability.

### CODE REVIEW

File: velox/common/encode/Base64.h [hunk 1]
```diff
@@ -32,125 +29,100 @@ class Base64 {
   static const size_t kCharsetSize = 64;
   static const size_t kReverseIndexSize = 256;
 
-  /// Character set used for encoding purposes.
-  /// Contains specific characters that form the encoding scheme.
+  /// Character set used for Base64 encoding.
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:32]: @Joe-Abraham why remove `/// Contains specific characters that form the encoding scheme.`?

File: velox/common/encode/Base64.h [hunk 2]
```diff
@@ -32,125 +29,100 @@ class Base64 {
...
   using Charset = std::array<char, kCharsetSize>;
 
-  /// Reverse lookup table for decoding purposes.
-  /// Maps each possible encoded character to its corresponding numeric value
-  /// within the encoding base.
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:41]: Why remove the other comments?

**Review by majetideepak — COMMENTED** [review]
@Joe-Abraham The header refactor seems to be missing some code comments.
Can you clarify? Thanks.

File: velox/common/encode/Base64.h [hunk 3]
```diff
@@ -32,125 +29,119 @@ class Base64 {
...
-  /// Decodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
-  /// returned by the calculateDecodedSize().
-  static void decode(const char* data, size_t size, char* output);
+  /// Encodes the input data using Base64 URL encoding.
+  static std::string encodeUrl(const char* input, size_t inputSize);
+  static std::string encodeUrl(folly::StringPiece text);
+  static std::string encodeUrl(const folly::IOBuf* inputBuffer);
+  static void
+  encodeUrl(const char* input, size_t inputSize, char* outputBuffer);
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:57]: `The outputBuffer must have enough /// space as returned by the calculateEncodedSize().` is missing?

File: velox/common/encode/Base64.h [hunk 4]
```diff
@@ -32,125 +29,119 @@ class Base64 {
...
-  static size_t calculateDecodedSize(const char* data, size_t& size);
-
-  /// Decodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
-  /// returned by the calculateDecodedSize().
-  static void decode(const char* data, size_t size, char* output);
+  /// Encodes the input data using Base64 URL encoding.
+  static std::string encodeUrl(const char* input, size_t inputSize);
+  static std::string encodeUrl(folly::StringPiece text);
+  static std::string encodeUrl(const folly::IOBuf* inputBuffer);
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:55]: Please retain the comments from before as they clarify the semantics.

Comment (Joe-Abraham) [reply velox/common/encode/Base64.h:58]: Reverted the changes.

File: velox/common/encode/Base64.h (hunk 3 above)
Comment (Joe-Abraham) [reply velox/common/encode/Base64.h:63]: Reverted the changes.

**Review by Joe-Abraham — COMMENTED** [review]
Updated the code according to the review comments.

**Review by majetideepak — APPROVED** [review]

File: velox/common/encode/Base64.h [hunk 5]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
-      const T& data,
+      const T& input,
       const Charset& charset,
-      bool include_pad,
-      char* out);
+      bool includePadding,
+      char* outputBuffer);
 
-  /// Decodes the specified data using the provided reverse lookup table.
+  // Encodes the specified data using the provided charset.
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:144]: This is `Decodes` correct?

Comment (majetideepak) [reply velox/common/encode/Base64.h:144]: Also must be `reverse lookup table`

Comment (Joe-Abraham) [reply velox/common/encode/Base64.h]: Correct it

File: velox/common/encode/Base64.h [hunk 6]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
-      len--;
+      inputSize--;
     }
     return numPadding;
   }
 
-  /// Performs a reverse lookup in the reverse index to retrieve the original
-  /// index of a character in the base.
-  static uint8_t base64ReverseLookup(char p, const ReverseIndex& reverseIndex);
+  // Reverse lookup helper function to get the original index of a Base64
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:125]: For multi-line comments in headers, we need `///`. 
https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md

File: velox/common/encode/Base64.h [hunk 7]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
-  static size_t
-  decode(const char* src, size_t src_len, char* dst, size_t dst_len);
+  /// Calculates the encoded size based on input size.
+  static size_t calculateEncodedSize(size_t inputSize, bool withPadding = true);
 
-  /// Decodes the specified number of characters from the 'src' using URL
-  /// encoding and writes the result to the 'dst'.
-  static void
-  decodeUrl(const char* src, size_t src_len, char* dst, size_t dst_len);
+  /// Calculates the decoded size based on encoded input and adjusts the input
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:102]: The previous comment was more descriptive.
```
/// Returns the actual size of the decoded data. Removes the padding
/// length from the input data 'inputSize'.
 ```

File: velox/common/encode/Base64.h [hunk 8]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
+      size_t outputSize);
 
-  /// Decodes the specified URL encoded text.
-  static std::string decodeUrl(folly::StringPiece text);
-
-  /// Decodes the specified number of characters from the 'src' and writes the
-  /// result to the 'dst'.
-  static size_t
-  decode(const char* src, size_t src_len, char* dst, size_t dst_len);
+  /// Calculates the encoded size based on input size.
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:99]: `... based on the input 'inputSize'.`

File: velox/common/encode/Base64.h [hunk 9]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
+      size_t outputSize);
+
+  /// Decodes the input Base64 URL encoded string.
+  static std::string decodeUrl(folly::StringPiece encodedText);
   /// Decodes the specified URL encoded payload and writes the result to the
   /// 'output'.
   static void decodeUrl(
       const std::pair<const char*, int32_t>& payload,
       std::string& output);
+  /// Decodes the specified number of characters from the 'input' using URL
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:91]: nit: new line above.

File: velox/common/encode/Base64.h [hunk 10]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
+  /// result to the 'outputBuffer'.
+  static size_t decode(
+      const char* input,
+      size_t inputSize,
+      char* outputBuffer,
+      size_t outputSize);
+
+  /// Decodes the input Base64 URL encoded string.
+  static std::string decodeUrl(folly::StringPiece encodedText);
   /// Decodes the specified URL encoded payload and writes the result to the
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:86]: new line above between functions.

File: velox/common/encode/Base64.h [hunk 11]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
+  static void decode(
+      const std::pair<const char*, int32_t>& payload,
+      std::string& output);
+  /// Decodes the specified number of characters from the 'data' and writes the
+  /// result to the 'output'. The output must have enough space as
+  /// returned by the calculateDecodedSize().
+  static void decode(const char* input, size_t inputSize, char* outputBuffer);
+  /// Decodes the specified number of characters from the 'input' and writes the
+  /// result to the 'outputBuffer'.
+  static size_t decode(
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:78]: new line between functions here and at other places.

File: velox/common/encode/Base64.h [hunk 12]
```diff
@@ -32,125 +29,125 @@ class Base64 {
...
+  encodeUrl(const char* input, size_t inputSize, char* outputBuffer);
 
+  /// Decodes the input Base64 encoded string.
+  static std::string decode(folly::StringPiece encodedText);
+  /// Decodes the specified encoded payload and writes the result to the
+  /// 'output'.
+  static void decode(
+      const std::pair<const char*, int32_t>& payload,
+      std::string& output);
+  /// Decodes the specified number of characters from the 'data' and writes the
```
Comment (majetideepak) [inline velox/common/encode/Base64.h:72]: `data` -> `input`, `output` -> `outputBuffer`.

File: velox/common/encode/Base64.cpp [hunk 13]
```diff
@@ -249,191 +254,198 @@ std::string Base64::encode(folly::StringPiece text) {
...
 }
 
 namespace {
 
 /**
- * this is a quick and dirty iterator implementation for an IOBuf so that the
- * template that uses iterators can work on IOBuf chains.  It only implements
- * postfix increment because that is all the algorithm needs, and it is a noop
- * since the read<>() function already incremented the cursor.
+ * This is a quick and simple iterator implementation for an IOBuf so that the
```
Comment (majetideepak) [inline velox/common/encode/Base64.cpp:264]: Let's fix this by using `//`.
https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md

**Review by majetideepak — COMMENTED** [review]
@Joe-Abraham found a few more changes. Can you also please rebase with the latest main? Thanks.

File: velox/common/encode/Base64.h (hunk 12 above)
Comment (Joe-Abraham) [reply velox/common/encode/Base64.h]: Updated the code

File: velox/common/encode/Base64.h (hunk 11 above)
Comment (Joe-Abraham) [reply velox/common/encode/Base64.h:87]: Updated the code

File: velox/common/encode/Base64.h (hunk 8 above)
Comment (Joe-Abraham) [reply velox/common/encode/Base64.h]: Updated the code

File: velox/common/encode/Base64.h (hunk 6 above)
Comment (majetideepak) [reply velox/common/encode/Base64.h:136]: I missed the private scope. This is fine. Only functions that need documentation need `///`

**Review by majetideepak — APPROVED** [review]
Thanks, @Joe-Abraham

**Comment by mbasmanova** [comment]
@Joe-Abraham Would you rebase?

**Comment by Joe-Abraham** [comment]
> @Joe-Abraham Would you rebase?

@mbasmanova I have rebased with master

File: velox/common/encode/Base64.h [hunk 14]
```diff
@@ -13,16 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 #pragma once
 
 #include <array>
-#include <exception>
```
Comment (mbasmanova) [inline velox/common/encode/Base64.h:19]: Not sure why, but I'm getting linter error that asks to move these after folly includes:

```
#include <array>
#include <string>

#include "velox/common/base/GTestMacros.h"
```

Any chance you can make that change?

Comment (Joe-Abraham) [reply velox/common/encode/Base64.h:19]: Updated the code.

**Pull request closed without merging**